server/utils_test.go

```diff
@@ -150,3 +150,26 @@ func TestRemoveDuplicatesFromSlice(t *testing.T) {
 		)
 	}
 }
+
+func TestSliceStringsMatch(t *testing.T) {
+	testCases := []struct {
+		a, b []string
+		want bool
+		name string
+	}{
+		{[]string{"foo", "bar"}, []string{"bar", "foo"}, true, "same elements in different order"},
+		{[]string{"foo", "bar"}, []string{"foo", "bar"}, true, "same elements in same order"},
+		{[]string{"foo", "bar"}, []string{"bar", "bar"}, false, "different number of same elements"},
+		{[]string{"foo", "bar"}, []string{"foo"}, false, "different lengths"},
+		{[]string{}, []string{}, true, "both slices empty"},
+		{[]string{"foo"}, []string{}, false, "one slice empty"},
+		{[]string{"unique"}, []string{"unique", "unique"}, false, "duplicate in one slice"},
```

Comment:
Maybe one case missing where both have duplicates, e.g. "foo", "foo" in both?

Reply:
good idea! pushed this, let me know if you had another case in mind.

```suggestion
+		{[]string{"foo", "foo", "bar"}, []string{"bar", "foo", "foo"}, true, "both have duplicates"},
+		{[]string{"foo", "bar", "bar"}, []string{"bar", "foo", "foo"}, false, "both have duplicates but elements don't match"},
```